pkg/ccl/changefeedccl/sink_cloudstorage.go

```diff
@@ -660,16 +660,21 @@ func (s *cloudStorageSink) flushTopicVersions(
 		return err == nil
 	})
 
-	// Files need to be cleared after the flush completes, otherwise file resources
-	for _, v := range toRemove {
-		s.files.Delete(cloudStorageSinkKey{topic: topic, schemaID: v})
-	}
-
 	// Allow synchronization with the async flusher to happen.
 	if s.testingKnobs != nil && s.testingKnobs.AsyncFlushSync != nil {
 		s.testingKnobs.AsyncFlushSync()
 	}
 
+	// Wait for the async flush to complete before clearing files.
+	err = s.waitAsyncFlush(ctx)
```

Comment:
Should we check for error before this line if flushFile has already returned an error (err is non-nil), but `waitAsyncFlush` doesn't return one? Is this possible? (If not, could we add comments to make this clearer) `flushFile` has an early return where we don't send the rest of the files if an error is returned early. We should probably still waitAsyncFlush if there is an error but return error after that. 

Not part of your code changes and perhaps we can follow up with a different PR to make this change more backportable. I’m finding the logic for handling cleanups and error returns varies across different code paths, and it’s becoming a bit confusing. Perhaps we could standardize the cleanup process a bit. For example, we are using `Delete` to remove files that have been flushed successfully and stop early if there is an error here. But for Flush, we just do s.Clear(true) if no error is returned. Do we have a case where we are not doing `s.Delete` on every files (which is same as s.Clear) if no error is returned from s.flushFile on every files?